source/slang/slang-emit-glsl.cpp

```diff
@@ -1171,32 +1261,11 @@ bool GLSLSourceEmitter::tryEmitInstExprImpl(IRInst* inst, const EmitOpInfo& inOu
             return false;
         }
         case kIROp_BitAnd:
+            return _tryEmitBitBinOp(inst, getInfo(EmitOp::BitAnd), getInfo(EmitOp::And), inOuterPrec);
         case kIROp_BitOr:
-        {
-            // Are we targetting GLSL, and are both operands scalar bools?
-            // In that case convert the operation to a logical And
-            if (as<IRBoolType>(inst->getOperand(0)->getDataType())
-                && as<IRBoolType>(inst->getOperand(1)->getDataType()))
-            {
-                EmitOpInfo outerPrec = inOuterPrec;
-                bool needClose = maybeEmitParens(outerPrec, outerPrec);
-
-                // Get the boolean version of the op
-                const auto op = _getBoolOp(inst->op);
-                auto prec = getInfo(op);
-
-                // TODO: handle a bitwise Or of a vector of bools by casting to
-                // a uvec and performing the bitwise operation
-
-                emitOperand(inst->getOperand(0), leftSide(outerPrec, prec));
-                m_writer->emit(prec.op);
-                emitOperand(inst->getOperand(1), rightSide(outerPrec, prec));
-
-                maybeCloseParens(needClose);
-                return true;
-            }
-            break;
-        }
+            return _tryEmitBitBinOp(inst, getInfo(EmitOp::BitOr), getInfo(EmitOp::Or), inOuterPrec);
+        case kIROp_BitXor:
```

Comment:
I know this is covered in you write up on the PR - but could do with a comment, as it is surprising.

Reply:
Good point. I will go back and comment this case, as well as document the declarations of the helper functions.